### PR TITLE
fix(typescript-operations): Generate ` $fragmentName` for fragment subtypes for fragment masking

### PR DESCRIPTION
**Pull request by izumin5210**
## Description

Related #7811 

`useFragment` for union was giving a type error because `$fragmentName` was not set to fragment subtypes when `inlineFragmenType: mask` was set.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Screenshots/Sandbox (if appropriate/relevant):

- https://codesandbox.io/s/gql-tag-operations-preset-fragment-masking-union-bug-2trm1w?file=/fixed.tsx:362-507
- https://codesandbox.io/s/gql-tag-operations-preset-fragment-masking-union-bug-2trm1w?file=/__generated__fixed__/graphql.ts:1923-2609

### before
image

### after
image

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/plugins/typescript/operations/tests/ts-documents.spec.ts [hunk 1]
```diff
@@ -5558,6 +5558,89 @@ function test(q: GetEntityBrandDataQuery): void {
...
+
+        union People = Character | Jedi | Droid
+
+        type Query {
+          people: People!
+        }
+      `);
+
+      const query = parse(/* GraphQL */ `
+        #graphql
```
Comment (n1ru4l) [inline packages/plugins/typescript/operations/tests/ts-documents.spec.ts:5585]:
```suggestion

```

Comment (n1ru4l) [reply packages/plugins/typescript/operations/tests/ts-documents.spec.ts:5585]: We don't need this comment here.

**Review by n1ru4l — CHANGES_REQUESTED** [review]
Thank you for this contribution, there is just one small suggested change, once that one is resolved we are ready to merge.

It would also be nice if you could update the dev-tests to also include fixtures for the cases you highlighted in the tests.

**Comment by izumin5210** [comment]
Thank you four your code review 😄 

I have added a new dev-test for complex cases ✅ 
👉  b0d6b8d

**Review by n1ru4l — APPROVED** [review]
Awesome 🎉

**Pull request merged**